src/BccCode.Linq/FilterInverter.cs

```diff
@@ -7,8 +7,7 @@ public static class FilterInverter
 {
     public static Filter Invert(Filter filter)
     {
-
-        var inverted = Activator.CreateInstance(filter.GetType(), "{}") as Filter;
+        var inverted = (Filter)Activator.CreateInstance(filter.GetType(), "{}");
```

Comment:
Why did you choose direct cast? Do we handle InvalidCastException somewhere?

Reply:
I did this to prevent possible null reference issues pointed out by Rider: In the further code the variable is uses like this: `inverted.Properties....`. This will result in a `NullReferenceException` exception. I thought it better to throw an exception here before later in code.   `NullReferenceException` exceptions are often quite difficult to examine.

Reply:
InvalidCastExceptions are not handled in the library - it needs to be handled by the user I think.